src/tiled/abstractobjecttool.cpp

```diff
@@ -246,6 +247,74 @@ void AbstractObjectTool::removeObjects()
     mapDocument()->removeObjects(mapDocument()->selectedObjects());
 }
 
+/**
+ * @brief AbstractObjectTool::applyCollisionsToSelection
+ *
+ * Add the selected collision masks for the currently selected tile - the last selected tile - to
+ * all selected tiles.
+ */
+void AbstractObjectTool::applyCollisionsToSelection()
+{
+    // The selected Collision Masks
+    QList<MapObject*> selectedObjects = mapDocument()->selectedObjects();
+
+    if (auto document = DocumentManager::instance()->currentDocument()) {
+        if (auto tilesetDocument = qobject_cast<TilesetDocument*>(document)) {
+            if (Tile *currentTile = dynamic_cast<Tile *>(tilesetDocument->currentObject())) {
```

Comment:
Avoid indenting the whole function by doing early-outs for these basic sanity checks:
```c++
Tile *currentTile = dynamic_cast<Tile *>(tilesetDocument->currentObject());
if (!currentTile)
    return;
```